src/model/guild/partial_guild.rs

```diff
@@ -450,4 +450,33 @@ impl PartialGuild {
     /// [Manage Webhooks]: permissions/constant.MANAGE_WEBHOOKS.html
     #[inline]
     pub fn webhooks(&self) -> Result<Vec<Webhook>> { self.id.webhooks() }
+
+    /// Obtain a reference to a role by its name.
+    ///
+    /// **Note**: If two or more roles have the same name, obtained reference will be one of them.
+    ///
+    /// # Examples
+    ///
+    /// Obtain a reference to a [`Role`] by its name.
+    ///
+    /// ```rust,no_run
+    /// use serenity::prelude::*;
+    /// use serenity::model::*;
+    /// struct Handler;
+    ///
+    /// use serenity::CACHE;
+    ///
+    /// impl EventHandler for Handler {
+    ///     fn on_message(&self, _: Context, msg: Message) {
+    ///         match message.guild_id().unwrap().get().unwrap().role_by_name("roll_name") {
```

Comment:
This can be simplified to an if-let, `if let Some(role) = message.guild_id()...`

Reply:
I'd also favour to use `find` instead of `get` here.